Amjad

MaysaM-M-Mousa opened the pull request.

Amjad-Mousa committed:
Some work in Flight & Flight Service

Amjad-Mousa committed:
Flight service

Amjad-Mousa committed:
Full code of Flight,FlightService

Amjad-Mousa committed:
Some Modification

MaysaM-M-Mousa commented on Airport-Managment-System/Entities/Flight.cs:
```diff
@@ -0,0 +1,139 @@
+            return flight;
+        }
+
+        public List<Seat> GetSeats()
+        {
+            return AvailableSeats;
+        }
```
why do you need this? you can get **AvailableSeats** directly since it uses the **set** in its declaration.

Amjad-Mousa replied:
done

MaysaM-M-Mousa commented:
There are almost 29 files should be ignored by `.gitignore` 😅

MaysaM-M-Mousa commented on Airport-Managment-System/Entities/Flight.cs:
```diff
@@ -0,0 +1,139 @@
+                   $"Seats: {seatsInfo}";
+        }
+
+        public List<Passenger> GetPassengers()
+        {
+            return this.Passengers;
+        }
```
same here

Amjad-Mousa replied:
done

MaysaM-M-Mousa commented on Airport-Managment-System/Entities/Flight.cs:
```diff
@@ -0,0 +1,139 @@
+        }
+    }
+
+    public class Seat
```
why not to extract this into its own class file?

Amjad-Mousa replied:
done

MaysaM-M-Mousa commented on Airport-Managment-System/Services/FlightService.cs:
```diff
@@ -0,0 +1,207 @@
+            {
+                Console.WriteLine($"Error reading CSV file: {ex.Message}");
+                this.flights = new List<Flight>(); 
+            }
```
since your source of truth is the csv file, I don't think you should swallow the exception and continue running the app in case of failure, the whole app should stop if there is something wrong in reading the csv file, so removing the try catch block makes more sense to me, what do u think?

Amjad-Mousa replied:
yes it was for catch the error that i was face (i remove it)

MaysaM-M-Mousa commented on Airport-Managment-System/Services/FlightService.cs:
```diff
@@ -0,0 +1,207 @@
+            }
+            catch (Exception ex)
+            {
+                return $"Error retrieving flight details: {ex.Message}";
```
Please avoid returning strings as failures, this will confuse the caller for these methods since it should determine the content of the string to know what is the operation result. just write the error in the console and return bool for example as a result for the operation result. Apply this please to other similar methods

Amjad-Mousa replied:
done

MaysaM-M-Mousa replied:
please check all methods in all services, I saw a punch of methods returning strings.

Amjad-Mousa replied:
done

Amjad-Mousa committed:
Update .gitignore

Amjad-Mousa committed:
ss

Amjad-Mousa committed:
ff

Amjad-Mousa committed:
Added modified files (excluding .vs, bin, and obj)

Amjad-Mousa committed:
Update .gitignore

Amjad-Mousa committed:
Add Booking,BookingService
Added modified files (excluding .vs, bin, and obj)

Amjad-Mousa committed:
Add Booking and BookingService

Amjad-Mousa committed:
modify some files

Amjad-Mousa committed:
Add Passenger.csv

Amjad-Mousa committed:
Modify Seats.csv

Amjad-Mousa committed:
Add a menu

Amjad-Mousa committed:
test

Amjad-Mousa committed:
remove "return strings"

Amjad-Mousa committed:
some edits

MaysaM-M-Mousa commented on Airport-Managment-System/Program.cs:
```diff
@@ -1,9 +1,209 @@
+        Console.Write("New Destination Country: ");
+        string destinationCountry = Console.ReadLine();
+
+        // Update other fields similarly...
```
u gonna do the rest?

Amjad-Mousa replied:
done

MaysaM-M-Mousa commented on Airport-Managment-System/Services/PassengerService.cs:
```diff
@@ -0,0 +1,37 @@
+        {
+            return CsvHelperService
+                .ReadFromCsv<Passenger>(@"../../../Data/Passenger.csv")
+                .Any(passenger => passenger.Id == passengerId);     
```
is not the data already loaded in the `passengers` list? why to read it again from the csv file?

Amjad-Mousa replied:
done

MaysaM-M-Mousa commented on Airport-Managment-System/Services/PassengerService.cs:
```diff
@@ -0,0 +1,37 @@
+                .Any(passenger => passenger.Id == passengerId);     
+        }
+
+        public string? GetPassengerDetails(int passengerId)
```
actually, it's expected to return an object here, this way the method will be more reusable and useful for different types of callers, the caller can do whatever with it (e.g convert it to string)

Amjad-Mousa replied:
we alreay have a method called GetPassengerById which returns a Passenger object 
so i change the name of this method to PrintPassengerDetails to be more clear

MaysaM-M-Mousa replied:
why to have a method for printing specifically? the caller can use `GetPassengerById` then print it as he wish right?

MaysaM-M-Mousa commented on Airport-Managment-System/Services/FlightService.cs:
```diff
@@ -0,0 +1,158 @@
+            return flights.FirstOrDefault(f => f?.Id == flightId);
+        }
+
+        public string GetFlightDetails(string id)
```
same comment here

Amjad-Mousa replied:
done

MaysaM-M-Mousa commented on Airport-Managment-System/Services/BookingService.cs:
```diff
@@ -0,0 +1,128 @@
+        {
+            this.Bookings = CsvHelperService.ReadFromCsv<Booking?>(csvFilePath) ?? new List<Booking?>();
+        }
+        public string GetBookingDetails(int BookingID)
```
same

Amjad-Mousa replied:
done

MaysaM-M-Mousa commented on Airport-Managment-System/Services/BookingService.cs:
```diff
@@ -0,0 +1,128 @@
+                booking.BookingClass = updatedBooking.BookingClass;
+            }
+
+            if (updatedBooking.BookingDate != default)
```
what if bookingDate is in the past? should u add some checks?

Amjad-Mousa replied:
done

MaysaM-M-Mousa commented on Airport-Managment-System/Services/BookingService.cs:
```diff
@@ -0,0 +1,128 @@
+
+        public bool DeleteBooking(int BookingID)
+        {
+            var flight = GetBookingById(BookingID);
```
type, the var name should be `booking` not `flight`

Amjad-Mousa replied:
done

MaysaM-M-Mousa commented on Airport-Managment-System/Services/BookingService.cs:
```diff
@@ -0,0 +1,128 @@
+                return false;   
+            }
+
+            if (Bookings.Any(b => b?.BookingId == booking.BookingId))
```
you could reuse `GetBookingById` function which u already have.

Amjad-Mousa replied:
done

Amjad-Mousa committed:
Refactor IsPassengerExists

Amjad-Mousa committed:
Rename GetPassengerDetails to PrintPassengerDetails

Amjad-Mousa committed:
Change method visibility and rename flight details method

Amjad-Mousa committed:
Change return type of PrintPassengerDetails method

Amjad-Mousa committed:
Rename GetBookingDetails to PrintBookingDetails

Amjad-Mousa committed:
Add validation for booking date to prevent past dates

Amjad-Mousa committed:
Fix variable name in DeleteBooking method

Amjad-Mousa committed:
Refactor booking existence check in BookingService

Amjad-Mousa committed:
Refactor UpdateFlight method for optional details

Enhanced the UpdateFlight method to allow optional updates for various flight details, including departure and arrival dates, airports, and pricing.
